### PR TITLE
Return metrics, Update docs

### LINKED ISSUE #839: bring back "--cls" : Specifying a list of specific classes to test on
### Search before asking

- [X] I have searched the YOLOv8 [issues](https://github.com/ultralytics/ultralytics/issues) and found no similar feature requests.

### Description

Hello,

I used to use the cls flag to specify a group of classes to test on, is there any way to do that ?

Thank you !

### Use case

Testing pretrained model on a specific global class. For example `-cls=[2, 3, 7]` (car, motorcycle, truck) to test on a vehicules dataset.

### Additional

_No response_

### Are you willing to submit a PR?

- [X] Yes I'd like to help by submitting a PR!

### PR DESCRIPTION
**Pull request by Laughing-q**
@AyushExel @glenn-jocher 
now the following usage is supported:
- detection
  ```python
  model = YOLO("yolov8s.pt")
  metrics = model.val(data="coco128.yaml")
  metrics.box.map    # map50-95
  metrics.box.map50  # map50
  metrics.box.map75  # map75
  metrics.box.maps   # a list contains map50-95 of each category
  ```
- segmentation
  ```python
  model = YOLO("yolov8s-seg.pt")
  metrics = model.val(data="coco128.yaml")
  metrics.box.map    # map50-95(B)
  metrics.box.map50  # map50(B)
  metrics.box.map75  # map75(B)
  metrics.box.maps   # a list contains map50-95(B) of each category
  metrics.seg.map    # map50-95(M)
  metrics.seg.map50  # map50(M)
  metrics.seg.map75  # map75(M)
  metrics.seg.maps   # a list contains map50-95(M) of each category
  ```
- classification
  ```python
  model = YOLO("yolov8n-cls.pt")
  metrics = model.val() 
  metrics.top1   # top1 accuracy
  metrics.top5   # top5 accuracy
  ```

## 🛠️ PR Summary

<sub>Made with ❤️ by [Ultralytics Actions](https://github.com/ultralytics/actions)<sub>

### 🌟 Summary
Enhancements and cleanup in training, evaluation, and documentation.

### 📊 Key Changes
- Removed `results` assignment when training models; calling `model.train()` is now an action without a return value.
- Introduced `metrics = model.val()` which now returns a metrics object containing evaluation results.
- Added `min_memory` configuration option to minimize memory usage during loss function computation.
- Introduced additional configuration parameters for detection and segmentation tasks in the form of `conf`, `iou`, `half` and more.
- Simplified `docs/cli.md` by removing Python code snippets and redundant CLI examples.
- Updated `docs/tasks/` documentation to reflect the new changes in metrics and model training/evaluation.
- `metrics` object now includes properties such as `map`, `map50`, `map75`, and a `maps` list for detailed evaluation results across different object detection and segmentation tasks.

### 🎯 Purpose & Impact
- The adjustment in the `model.train()` method simplifies the training process interface, reducing confusion about its usage.
- Returning a metrics object in `model.val()` enhances the clarity and accessibility of model evaluation data for further analysis.
- The `min_memory` option allows users with memory constraints to train models more effectively.
- New configuration parameters and clearer documentation improve user understanding and control over model training and inference.
- Cleaner and more focused CLI documentation makes it easier for users to utilize command-line interfaces for tasks.
- Updated task-specific documentation in `docs/tasks/`, like classification, detection, and segmentation, ensures users have the latest information for using YOLO models.
- Enhanced metrics provide users with more detailed and granular insight into model performance, ensuring better assessment and decision-making based on evaluation results.

### CODE REVIEW

**Comment by AyushExel** [comment]
Looks good!

**Comment by glenn-jocher** [comment]
@Laughing-q this is awesome! I like it.

**Comment by glenn-jocher** [comment]
PR merged!
